code/game/objects/items/grenades/discogrenade.dm

```diff
@@ -0,0 +1,128 @@
+//Ethereal Disco Grenade for Ethereal traitors
+//Does not affect ethereals.
+//Some basic code peices taken from flashbang, spawner grenade and ethereal disco ball for functionality (basically a combination of the 3).
+
+//////////////////////
+// Primary grenade  //
+//////////////////////
+
+/obj/item/grenade/discogrenade
+	name = "Ethereal Disco Grenade"
+	desc = "An unethical micro-party that will make all non-Ethereal beings dance to its beat!"
+	icon_state = "disco"
+	item_state = "disco"
+	lefthand_file = 'icons/mob/inhands/equipment/security_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/security_righthand.dmi'
+	var/list/messages = list("This party is great", "Wooo!!!", "Party!", "Check out these moves!", "Hey, want to dance with me?")
+	var/list/message_social_anxiety = list("I want to go home...", "Where are the toilets?", "I don't like this song")
```

Comment:
add a period or an ellipsis to the end of "I don't like this song"

Reply:
Done